### PR TITLE
Issue3031 add SnapshotCache to overflow to disk

### LINKED ISSUE #3031: Provide Snapshotcache that can overflow to (local) disk

### PR DESCRIPTION
**Pull request by uweschaefer**
Closes #3031

### CODE REVIEW

File: factcast-site/documentation-docsy/content/en/Setup/properties.md [hunk 1]
```diff
@@ -53,21 +53,26 @@ description: Properties you can use to configure FactCast
...
+
+#### InMemAndDisk-Snapshots
+
+| Property                            | Description                                                                                                                                                                                           | Default        |
+| ----------------------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :------------- |
+| factcast.local.disk.pathToSnapshots | path to store the snapshots in the file system.                                                                                                                                                       | java.io.tmpdir |
+| factcast.local.disk.maxDiskSpace    | max disk space to be used by the SnapshotDiskRepository. The oldest Snapshots will start to be removed after reaching 90% of the allocated space, never reaching 100%. Use 0 for unlimited disk space | 0              |
+
```
Comment (bresciamattia) [inline factcast-site/documentation-docsy/content/en/Setup/properties.md:68]: also this should change if we align it with https://github.com/factcast/factcast/issues/3035

File: factcast-snapshotcache-local-disk/src/main/java/org/factcast/core/snap/local/InMemoryAndDiskSnapshotProperties.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@Data
+@Slf4j
+@Accessors(fluent = false)
+@ConfigurationProperties(prefix = InMemoryAndDiskSnapshotProperties.PROPERTIES_PREFIX)
+public class InMemoryAndDiskSnapshotProperties {
+  public static final String PROPERTIES_PREFIX = "factcast.local.disk";
```
[the remark this answers is not in the document]
Comment (mamup11) [reply factcast-snapshotcache-local-disk/src/main/java/org/factcast/core/snap/local/InMemoryAndDiskSnapshotProperties.java]: Updated 👍

File: factcast-site/documentation-docsy/content/en/Setup/properties.md (hunk 1 above)
Comment (mamup11) [reply factcast-site/documentation-docsy/content/en/Setup/properties.md]: Same here

File: factcast-snapshotcache-local-disk/src/main/java/org/factcast/core/snap/local/SnapshotFileHelper.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    String withSlashes = addSlashes(hash);
+    return new File(persistenceDirectory, withSlashes);
+  }
+
+  String addSlashes(String hash) {
+    return new StringBuilder(hash)
+        .insert(16, '/')
+        .insert(12, '/')
+        .insert(8, '/')
+        .insert(4, '/')
```
Comment (bresciamattia) [inline factcast-snapshotcache-local-disk/src/main/java/org/factcast/core/snap/local/SnapshotFileHelper.java:51]: same as above

**Review by bresciamattia — COMMENTED** [review]
@mamup11 overall lgmt! Just a couple of small comments. Please also take a second look at the sonar report as well, I think that some of those code smells can be tackled easily.

**Review by bresciamattia — APPROVED** [review]

**Pull request merged**